src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,165 @@
+
+/-- Transfinite induction construction of the sigma-algebra generated by a set of sets `s`. At each
+step, we add all elements of `s`, the empty set, the complements of already constructed sets, and
+countable unions of already constructed sets. We index this construction by an ordinal `< ω₁`, as
+this will be enough to generate all sets in the sigma-algebra. -/
+def generate_measurable_rec (s : set (set α)) : ω₁ → set (set α)
+| i := let S := ⋃ j : {j // j <₁ i}, generate_measurable_rec j.1 in
+    s ∪ {∅} ∪ compl '' S ∪ (set.range (λ (f : ℕ → S), ⋃ n, (f n).1))
+using_well_founded {dec_tac := `[exact j.2]}
+
+/-- At each step of the inductive construction, the cardinality bound `≤ (max (#s) 2) ^ ω`
+holds. -/
+lemma cardinal_generate_measurable_rec_le (s : set (set α)) (i : ω₁) :
+  #(generate_measurable_rec s i) ≤ (max (#s) 2) ^ omega.{u} :=
+begin
+  apply (aleph 1).ord.out.wo.wf.induction i,
+  assume i IH,
+  have A := omega_le_aleph 1,
+  have B : aleph 1 ≤ (max (#s) 2) ^ omega.{u} :=
+    aleph_one_le_continuum.trans (power_le_power_right (le_max_right _ _)),
+  have C : omega.{u} ≤ (max (#s) 2) ^ omega.{u} := A.trans B,
+  have J : #(⋃ (j : {j // j <₁ i}), generate_measurable_rec s j.1) ≤ (max (#s) 2) ^ omega.{u},
+  { apply (mk_Union_le _).trans,
+    have D : # {j // j <₁ i} ≤ aleph 1 := (mk_subtype_le _).trans (le_of_eq (aleph 1).mk_ord_out),
+    have E : cardinal.sup.{u u}
+      (λ (j : {j // j <₁ i}), #(generate_measurable_rec s j.1)) ≤ (max (#s) 2) ^ omega.{u} :=
+    cardinal.sup_le.2 (λ ⟨j, hj⟩, IH j hj),
+    apply (mul_le_mul' D E).trans,
+    rw mul_eq_max A C,
+    exact max_le B le_rfl },
+  rw [generate_measurable_rec],
+  apply_rules [(mk_union_le _ _).trans, add_le_of_le C, mk_image_le.trans],
+  { calc #s ≤ max (#s) 2 : le_max_left _ _
+    ... ≤ (max (#s) 2) ^ omega.{u} : self_le_power _ one_lt_omega.le },
+  { rw [mk_singleton],
+    exact one_lt_omega.le.trans C },
+  { apply mk_range_le.trans,
+    simp only [mk_pi, subtype.val_eq_coe, prod_const, lift_uzero, mk_denumerable, lift_omega],
+    have := @power_le_power_right _ _ omega.{u} J,
+    rwa [← power_mul, omega_mul_omega] at this }
+end
+
+lemma cardinal_Union_generate_measurable_rec_le (s : set (set α)) :
+  #(⋃ i, generate_measurable_rec s i) ≤ (max (#s) 2) ^ omega.{u} :=
+begin
+  apply (mk_Union_le _).trans,
+  rw [(aleph 1).mk_ord_out],
+  refine le_trans (mul_le_mul' aleph_one_le_continuum
+    (cardinal.sup_le.2 (λ i, cardinal_generate_measurable_rec_le s i))) _,
+  have := power_le_power_right (le_max_right (#s) 2),
+  rw mul_eq_max omega_le_continuum (omega_le_continuum.trans this),
+  exact max_le this le_rfl
+end
+
+/-- A set in the sigma-algebra generated by a set of sets `s` is constructed at some step of the
+transfinite induction defining `generate_measurable_rec`.
+The other inclusion is also true, but not proved here as it is not needed for the cardinality
+bounds.-/
+theorem generate_measurable_subset_rec (s : set (set α)) ⦃t : set α⦄
```

Comment:
Just one last thing actually, `s` can be inferred here.

Reply:
Not in the application below, where `t` is kept implicit, as well as `ht`.